media-tv/v4l-dvb-saa716x/v4l-dvb-saa716x-0.0.1_p20170225-r2.ebuild

```diff
@@ -52,9 +53,12 @@ src_prepare() {
 }
 
 src_compile() {
-	BUILD_PARAMS="SUBDIRS=${S}/linux/drivers/media/common/saa716x \
-	CONFIG_SAA716X_CORE=m CONFIG_DVB_SAA716X_FF=m CONFIG_DVB_SAA716X_BUDGET=m \
-	CONFIG_DVB_SAA716X_HYBRID=m"
-	addpredict "${EROOT%/}"/usr/src/linux/
+	kernel_is le 5 2 && BUILD_PARAMS="SUBDIRS=${S}/linux/drivers/media/common/saa716x \
+		CONFIG_SAA716X_CORE=m CONFIG_DVB_SAA716X_FF=m CONFIG_DVB_SAA716X_BUDGET=m \
+		CONFIG_DVB_SAA716X_HYBRID=m"
+	kernel_is gt 5 2 && BUILD_PARAMS="M=${S}/linux/drivers/media/common/saa716x/ \
+		CONFIG_SAA716X_CORE=m CONFIG_DVB_SAA716X_FF=m CONFIG_DVB_SAA716X_BUDGET=m \
+		CONFIG_DVB_SAA716X_HYBRID=m"
```

Comment:
Could this be a bit more concise - I don't see any other differences between the BUILD_PARAMS e.g.:

```sh
	kernel_is le 5 2 && BUILD_PARAMS="SUBDIRS" || BUILD_PARAMS="M"
	BUILD_PARAMS+="=${S}/drivers/media/pci/saa716x \
```

Reply:
Yes, good idea.
I tested a little an figured out that `BUILD_PARAMS="M"` works from kernel 5.1 up - will change that, too.